pkg/solana/txm/txm.go

```diff
@@ -505,6 +514,82 @@ func (txm *Txm) handleErrorSignatureStatus(sig solanaGo.Signature, status *rpc.S
 	}
 }
 
+// handleReorg detects and manages transaction state regressions (re-orgs) for a given signature.
+//
+// A re-org occurs when the on-chain state of a signature regresses:
+// - Confirmed -> Processed || Not Found
+// - Processed -> Not Found
+//
+// This function determines if the signature’s state regression impacts the overall transaction state and, if so, takes appropriate action:
+//   - For regressions from "Confirmed", our in memory layer is updated and the tx is rebroadcasted with a new hash restarting the retry/bumping cycle.
+//   - For regressions from "Processed", the existing retry/bumping cycle is still running, so no immediate action is needed. We only update our in-memory state to Broadcasted.
+//     Future rebroadcasts, will be handled by the TxExpirationRebroadcast logic (if enabled) when the transaction expires.
+func (txm *Txm) handleReorg(ctx context.Context, client client.ReaderWriter, sig solanaGo.Signature, status *rpc.SignatureStatusesResult) {
+	// Retrieve the last known status of the transaction associated with this signature from the in-memory layer.
+	txInfo, err := txm.txs.GetSignatureInfo(sig)
+	if err != nil {
+		txm.lggr.Errorw("failed to get signature info when checking for potential re-orgs", "signature", sig, "error", err)
+		return
+	}
+
+	// Check if the sig status has regressed to indicate a re-org.
+	// A regression is identified when the state transitions as follows:
+	// - Confirmed -> Processed || Broadcasted || Not Found
+	// - Processed -> Broadcasted || Not Found
+	currentTxState := convertStatus(status)
+	if regressionType, isRegressed := isStatusRegression(txInfo.state, currentTxState); isRegressed {
+		if err := txm.txs.UpdateSignatureStatus(sig, currentTxState); err != nil {
+			txm.lggr.Errorw("failed to update sig status", "signature", sig, "error", err)
+			return
+		}
+
+		// Determine if the sig regression affects the transaction state.
+		// If the tx isn't considered re-orged, skip further processing.
+		// Multiple signatures may be in-flight for a single transaction, so a re-org
+		// for one signature doesn't necessarily mean the transaction state has regressed.
+		if !txm.txs.TxHasReorg(txInfo.id) {
+			return
+		}
+
+		txm.lggr.Warnw("re-org detected for transaction", "txID", txInfo.id, "signature", sig, "previousStatus", txInfo.state, "currentStatus", currentTxState)
```

Comment:
Think we can consider re-orgs as part of the normal TXM processes since it's an expected occurrence eventually. `Warn` might be too alarming here. I think `Info` or even `Debug` might be a better log level here.

Reply:
Actually let's do `Debug` to keep this the same as our expiration log